proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -2195,11 +2064,17 @@ lemma (in delete_one_conc) suspend_invs'[wp]:
    apply (clarsimp simp: sym_refs_asrt_def)
   apply (rule hoare_seq_ext_skip, wpsimp)
   apply (rule hoare_seq_ext_skip)
-  apply clarsimp
+   apply clarsimp
    apply (wpsimp simp: updateRestartPC_def)
-  apply (wpsimp wp: sts_tcbSchedDequeue_invs' tcbReleaseRemove_invs'
-                    schedContextCancelYieldTo_invs')
-  apply (fastforce simp: tcb_st_refs_of'_def st_tcb_at'_def obj_at'_def projectKOs)
+  apply (rule_tac B="\<lambda>_ s. invs' s
+                           \<and> sch_act_simple s
+                           \<and> tcb_at' t s
+                           \<and> t \<noteq> ksIdleThread s
+                           \<and> st_tcb_at' ((=) Inactive) t s"
+               in hoare_seq_ext[rotated])
```

Comment:
Very minor, but if these manual instantiations are definitely needed then you could reduce some duplication by giving a name to the original precondition in the lemma statement.

Reply:
Good spotting. Done!